### DISCUSSION TITLE
MemoryError when running Caiman many times

### DISCUSSION THREAD

**Discussion by zhouyi0812** [flatironinstitute/CaImAn#1271] [Q&A] [answered]
Hello!
Every few times after running caiman, there will be an error: MemoryError: Unable to allocate 3.56 GiB for an array with shape (540, 1152, 1536) and data type float32
And the number of GiB require varies. I usually restart the computer and the issue fix.
I wonder if there is a code that I can clean the cache on caiman so I don't have to restart the computer, and if it is related to running many times?
Thank you!

**Comment by kushalkolar** [comment, marked as answer]
You can probably just restart the kernel and don't need to reboot.

Proper garbage collection in jupyter can be very complicated even for advanced programmers, due to the ipython kernel making tons of new references. Restarting the kernel is the way to go in this situation.
